### ISSUE TITLE
sql: panic when calling cluster_logical_timestamp after an TransactionAbortedError

### ISSUE THREAD

**Issue by andreimatei** [cockroachdb/cockroach#8455]
So rare that what I think will happen actually happens...

`MinInitialTimestamp` is set only on in the `!inTxn` case, which means it's not set after a user-controlled transaction restart.
https://github.com/cockroachdb/cockroach/blob/c536df4a669fdeb8511114b53536fd6e25b022f5/sql/executor.go#L467

In particular, it's not set after a restart caused by a `TransactionRestartedError`. After these errors, the `Transaction` proto is cleared, so the `OrigTimestamp` is wiped, which causes `planner.SetTxn()` to do `p.evalCtx.SetClusterTimestamp(txn.Proto.OrigTimestamp)`, which then causes the `cluster_logical_timestamp()` function to panic.

Note that, if the transaction is auto-retried, it no longer crashes, because there the different retries keep resetting the `OrigTimestamp` in the proto to `opt.MinInitialTimestamp`. But that's no bueno for a different reason, which I'll file separately.

To repro:

session 1:
```
root@:26257> begin; savepoint COCKROACH_RESTART;
OK
root@:26257> insert into x(a,b) values (1,2);
INSERT 1
```

session 2:
```
root@:26257> begin priority high; delete from x; commit;
```

session 1 again:
```
root@:26257> RELEASE SAVEPOINT COCKROACH_RESTART;
pq: restart transaction: txn aborted "sql/executor.go:477 sql txn" id=440a21ce key=/Table/52/1/166451967745785857/0 rw=true pri=0.45511574 iso=SERIALIZABLE stat=ABORTED epo=0 ts=1470867512.422560116,1 orig=1470867505.797398689,0 max=1470867506.047398689,0 wto=false rop=false
root@:26257> select cluster_logical_timestamp();
pq: current transaction is aborted, commands ignored until end of transaction block
root@:26257> ROLLBACK TO SAVEPOINT COCKROACH_RESTART;
OK
root@:26257> select cluster_logical_timestamp();
#PANIC
```

**Comment by knz** [comment]
This will be fixed when #8456 is addressed. See my comment in the other issue for a strategy for a fix.

I'm unassigning this since I'm not working on it until next week. Feel free to re-assign if it can wait until then.

**Comment by knz** [comment]
Hmm I can't seem to repro this issue using the example given. Could we have a look at this together?

**Comment by knz** [comment]
Closed via #8596.
